### PR TITLE
Carets- Julia Meier- API Muncher

### PR DESCRIPTION
**Pull request by julmeier**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I experimented with Postman to query the API. It also helped to read through the documentation. Once creating my api_wrapper class and recipe class, I experimented some more on how to best query the API. |
| Describe your API Wrapper.  How did you decide on the methods you created? | The API Wrapper class has methods that constructs the API url needed to return either a collection of recipes or a single recipe, and corresponding methods to create ruby objects from the received API recipe data. I also have a method that just returns the number of recipes in the Edamam API for a give keyword search. |
| Describe an edge case or failure case test you wrote for your API Wrapper. | One test makes sure that a bogus request (ex. "jklmnopqrstuv" in the search bar) will return an empty array. |
| Explain how VCR aids in testing an API. | A VCR records an application's interaction with an HTTP request. It is helpful in testing APIs because the interaction can be called and recorded only once. Tests can use this recording without making more calls to the API. This is helpful if there is a limit on the number of API calls (for example, Edamam only allows 5 per minute, which would make controller testing difficult) or the API calls are expensive. It also allows programmers to test offline. |
| What is the Heroku URL of your deployed application? | http://hidden-woodland-83927.herokuapp.com/ |

### CODE REVIEW

File: test/controllers/recipes_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      it "redirects to the home page if the user enters nothing as a keyword" do
+        get recipes_search_path params:{keywords: ""}
+        must_redirect_to root_path
+        flash[:failure].must_equal "You did not enter any keywords. Please enter ingredient keywords into the search bar."
+      end
+    end
+
+    describe "search by page (i.e. pagination)" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:29]: Good sets of tests.

File: test/controllers/recipes_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        VCR.use_cassette("recipes") do
+          get recipes_search_path params:{keywords: "lemon chicken"}
+          must_respond_with :success
+        end
+      end
+
+      it "redirects to the home page if the user enters nothing as a keyword" do
+        get recipes_search_path params:{keywords: ""}
+        must_redirect_to root_path
+        flash[:failure].must_equal "You did not enter any keywords. Please enter ingredient keywords into the search bar."
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:25]: Your controller doesn't actually add that flash message.

File: app/controllers/recipes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  #THIS IS THE SEARCH PAGE WITH QUERY BAR
+  def index
+  end
+
+  #THIS DISPLAYS THE RESULTS OF WHAT WE FOUND WHEN WE DID THE SEARCH
+  def search #recipes_search_path
+      #binding.pry
+      @keywords = params['keywords']
+      if @keywords.blank?
+        redirect_to root_path
```
Comment (CheezItMan) [inline app/controllers/recipes_controller.rb:15]: no flash notice?

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Lots of commits, good mesages |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) |  |
| Semantic HTML | Reasonable semantic HTML, you do have some `div` elements which are there for styling, that could be replaced or dropped and the semantic tags used instead.. |
| Errors are reported to the user | Flash notices if they didn't search for anything. |
| API Wrapper to handle the API requests | Check |
| Controller testing | Tests written, but not working,  You did have the postgres problem earlier.   |
| Lib testing | Check |
| Search Functionality | Work great!  |
| List Functionality | Well done!  |
| Show individual item functionality (link to original recipe opens in new tab) | This is **missing**. |
| **Styling** |  |
| Foundation Styling for responsive layout | Check |
| List View shows 10 items at a time/pagination | Well done |
| The app is styled to create an attractive user interface | Nicely styled |
| **API Features** | |
| The App attributes Edaman | Check |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to Trello Board | NA |
| Link to deployed app on Heroku | Well done |
|  **Overall** |  Nicely done, you recovered well from the Postgres problem after I showed you how to fix it.  Pretty good testing and nice styling.  Well done. |
